### PR TITLE
Add mongoose examples for HTTP, HTTPS, AWS-IOT MQTT, WebSocket

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

File: samples/net/mongoose/.clang-format [hunk 1]
```diff
@@ -0,0 +1,6 @@
+BasedOnStyle: Google
```
Comment (rlubos) [inline samples/net/mongoose/.clang-format:1]: Any sample should follow Zephyr coding guidelines.

File: samples/net/mongoose/http-server/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# Mail: support@cesanta.com
+#
+# SPDX-License-Identifier: GPL2.0 or commercial
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(mongoose_http_server)
+
+#add_definitions(-DMG_ENABLE_LINES=1)
+add_definitions(-DMG_ENABLE_SSI=0)
```
Comment (rlubos) [inline samples/net/mongoose/http-server/CMakeLists.txt:11]: Any sample-specific configurations sholud rather be Kconfig entries.

File: samples/net/mongoose/http-server/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+add_definitions(-I${CMAKE_CURRENT_SOURCE_DIR}/../common)
```
Comment (rlubos) [inline samples/net/mongoose/http-server/CMakeLists.txt:12]: I think `target_include_directories()` is a better fit here.

File: samples/net/mongoose/http-server/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+add_definitions(-g3 -O0)
```
Comment (rlubos) [inline samples/net/mongoose/http-server/CMakeLists.txt:13]: Please don't change compiler options that way, use Kconfig configuration instead.

File: samples/net/mongoose/common/mongoose.c [hunk 5]
```diff
@@ -0,0 +1,5115 @@
...
+//
+// You are free to use this software under the terms of the GNU General
+// Public License, but WITHOUT ANY WARRANTY; without even the implied
+// warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU General Public License for more details.
+//
+// Alternatively, you can license this software under a commercial
+// license, as set out in https://www.mongoose.ws/licensing/
+//
+// SPDX-License-Identifier: GPL-2.0 or commercial
```
Comment (rlubos) [inline samples/net/mongoose/common/mongoose.c:18]: I think this is a serious no-go for this PR, we can't accept GPL licensed code into Zephyr. CC @carlescufi

Comment (carlescufi) [reply samples/net/mongoose/common/mongoose.c:18]: Yep, unfortunately we cannot accept GPL code in Zephyr. Can you please find an alternative?

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**